web/src/LocalStorage.tsx

```diff
@@ -0,0 +1,63 @@
+import React from "react"
+
+export type LocalStorageContext = {
+  set: (key: string, value: any) => void
+  get: <T extends {}>(key: string) => T | null
+}
+
+export const localStorageContext = React.createContext<LocalStorageContext>({
+  set: (key: string, value: any): void => {},
+  get: <T extends {}>(key: string): T | null => null,
+})
+
+export function makeKey(tiltfileKey: string, key: string): string {
+  return "tilt-" + JSON.stringify({ tiltfileKey: tiltfileKey, key: key })
+}
+
+// Provides access localStorage, but namespaced by `tiltfileKey`
+// Also handles serialization and typing.
+export function LocalStorageContextProvider(
+  props: React.PropsWithChildren<{ tiltfileKey?: string }>
+) {
+  let context: LocalStorageContext
+
+  if (!props.tiltfileKey) {
+    // if there's no tiltfile key (presumably because we're initializing), just make this all a noop
```

Comment:
it might be be better to error out here.

HUD short-circuits to a load screen if it's initializing https://github.com/tilt-dev/tilt/blob/master/web/src/HUD.tsx#L246 so you don't need to worry about this case, and i'm not even sure this would work correctly

Reply:
or even better, make the arg non-optional!